tsl/src/bgw_policy/job.c

```diff
@@ -618,12 +655,9 @@ policy_recompression_execute(int32 job_id, Jsonb *config)
 			 "completed recompressing chunk \"%s.%s\"",
 			 NameStr(chunk->fd.schema_name),
 			 NameStr(chunk->fd.table_name));
+		CommitTransactionCommand();
 	}
 
-	chunkid = get_chunk_to_recompress(dim, config);
-	if (chunkid != INVALID_CHUNK_ID)
-		enable_fast_restart(job_id, "recompression");
-
```

Comment:
i like that change, this is not the only bgw job with that logic so maybe we should adjust similar jobs in a followup pr